code/modules/telecomms/telecomms.dm

```diff
@@ -0,0 +1,276 @@
+
+	if(!src.isUsable())
+		return 0
+
+	broadcastMessage(source, mobName, sayVerb, message, freq, speaking)
+	return 1
+
+/obj/machinery/telecomms_machine/proc/isUsable()
+	if(!src.on)
+		return 0
+	if(stat & (EMPED|NOPOWER|BROKEN))
+		return 0
+	return 1
+
+/proc/broadcastMessage(var/mob/living/source, var/mobName, var/sayVerb, var/message, var/freq, var/datum/language/speaking, var/interceptData)
+	//world << "[source], [mobName], [sayVerb], [message], [freq], [speaking], [interceptData]"
+	var/list/radios = list()
+
+	for(var/obj/item/device/radio/R in all_radios)
+		if(R.receive_range(freq) != -1)
+			radios += R
+
+	var/list/receive = get_mobs_in_radio_ranges(radios)
+
+	var/list/heard_normal 	= list() // normal message
+	var/list/heard_voice 	= list() // voice message	(ie "chimpers")
+	var/list/heard_garbled	= list() // garbled message (ie "f*c* **u, **i*er!")
+
+	for(var/mob/R in receive)
+		if(R.client)
+			if(R.client.prefs)
+				if(!(R.client.prefs.toggles & CHAT_RADIO)) //Adminning with 80 people on can be fun when you're trying to talk and all you can hear is radios.
+					continue
+			else
+				log_debug("Client prefs found to be null in /proc/broadcastMessage() for mob [R] and client [R.ckey], this should be investigated.")
+
+		if(istype(R, /mob/new_player)) // we don't want new players to hear messages. rare but generates runtimes.
+			continue
+
+		if(interceptData && istype(R, /mob/dead/observer) && R.client && R.client.prefs && (R.client.prefs.toggles & CHAT_GHOSTRADIO)) // Don't spam ghosts with double messages
+			continue
+
+		if (!source || R.say_understands(source))
+			heard_normal += R
+		else
+			if(source && source.speak_emote.len)
+				heard_voice += R
+
+			else
+				heard_garbled += R
+
+	var/freq_text = get_frequency_name(interceptData ? interceptData : freq)
+	if(length(heard_normal) || length(heard_voice) || length(heard_garbled))
+
+		var/part_b_extra = ""
+		if(interceptData) // intercepted radio message
+			part_b_extra = " <i>(Intercepted)</i>"
+		var/part_a = "<span class='[frequency_span_class(interceptData ? interceptData : freq)]'><b>\[[freq_text]\][part_b_extra]</b> <span class='name'>" // goes in the actual output
+
+		var/part_b = "</span> <span class='message'>" // Tweaked for security headsets -- TLE
+
+		if(length(heard_normal))
+			for(var/mob/R in heard_normal)
+				R.hear_radio(message, sayVerb, speaking, part_a, part_b, source, 0, mobName)
+
+		if(length(heard_voice))
+			for(var/mob/R in heard_voice)
+				R.hear_radio(message, sayVerb, speaking, part_a, part_b, source, 0, mobName)
+
+		if(length(heard_garbled))
+			for(var/mob/R in heard_garbled)
+				R.hear_radio(message, sayVerb, speaking, part_a, part_b, source, 1, mobName)
+
+	if(!interceptData && !(freq in ANTAG_FREQS))
+		for(var/antFreq in ANTAG_FREQS)
+			broadcastMessage(source, mobName, sayVerb, message, antFreq, speaking, freq)
+
+	return 1
+
+
+// Round up // REMIND ME TO MOVE THIS TO A BETTER PLACE // Actually nevermind, it's fine
```

Comment:
It's not:
https://github.com/PolarisSS13/Polaris/blob/1e62406b6bce80e2364872bd9ec41ff5b79e9930/code/_helpers/maths.dm#L3
But other things use it so removing it would be better left to another PR.
Not to mention it'll return wrong result on integers (since it always assumes remainder)